src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -634,7 +654,7 @@ private long getOffset()
     {
       if (fileNames.isEmpty())
       {
-        return committedOffset.get();
+        return committedOffset.get() + 1;
```

Comment:
This function needs to return the offset that is going to be sent for next round. For example, if we already committed record 4, we should return 5 here, so that the next record we receive will be record  5.

Reply:
Previously committed and flushed offset stores a + 1. I think we discussed that either we change the naming to nextCommittedOffset or we change committedOffset to store "a" instead of "a + 1". I chose the second option. But this function requires retuning "a+1", that's why I have this plus one added in return.